src/openrct2-ui/scripting/ScWindow.hpp

```diff
@@ -229,38 +229,37 @@ namespace OpenRCT2::Scripting
             return result;
         }
 
-        std::vector<int32_t> colours_get() const
+        std::vector<DukValue> colours_get() const
         {
-            std::vector<int32_t> result;
+            std::vector<DukValue> result;
             auto w = GetWindow();
             if (w != nullptr)
             {
+                auto apiVersion = GetTargetAPIVersion();
+                auto ctx = GetContext()->GetScriptEngine().GetContext();
                 result.reserve(std::size(w->colours));
                 for (auto c : w->colours)
```

Comment:
Currently this seems to return an array of 6 colours, while I have not found any instance of windows using more than the first 3 colours. Would it be possible to only return the first 3 colours?

Reply:
Possible yes, desirable probably not. Also note that it already returns 6 colours, there is no change there.